public/pages/utils/anomalyResultUtils.ts

```diff
@@ -1179,3 +1253,164 @@ export const parseEntityAnomalySummaryResults = (
+              },
+            },
+          },
+          {
+            term: {
+              task_id: taskId,
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      [MIN_END_TIME]: {
+        min: {
+          field: AD_DOC_FIELDS.DATA_END_TIME,
+        },
+      },
+      [MAX_END_TIME]: {
+        max: {
+          field: AD_DOC_FIELDS.DATA_END_TIME,
+        },
+      },
+    },
+  };
+};
+
+export const getHistoricalAggQuery = (
+  startTime: number,
+  endTime: number,
+  taskId: string,
+  anomalyAgg: ANOMALY_AGG
+) => {
+  // Need to calculate timezone offset before bucketing into days/months/weeks.
+  const timezoneAsIANAString = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  return {
+    size: MAX_ANOMALIES,
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              [AD_DOC_FIELDS.ANOMALY_GRADE]: {
+                gt: 0,
+              },
+            },
+          },
+          {
+            range: {
+              data_end_time: {
+                gte: startTime,
+                lte: endTime,
+              },
+            },
+          },
+          {
+            term: {
+              task_id: taskId,
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      [AGGREGATED_ANOMALIES]: {
+        date_histogram: {
+          field: AD_DOC_FIELDS.DATA_END_TIME,
+          calendar_interval: anomalyAgg,
+          time_zone: timezoneAsIANAString,
+          extended_bounds: {
+            min: startTime,
+            max: endTime,
+          },
+        },
+        aggs: {
+          [MAX_ANOMALY_AGGS]: {
+            max: {
+              field: AD_DOC_FIELDS.ANOMALY_GRADE,
+            },
+          },
+        },
+      },
+    },
+  };
+};
+
+export const parseHistoricalAggregatedAnomalies = (
+  result: any,
+  anomalyAgg: ANOMALY_AGG
+) => {
+  const resultBuckets = get(
+    result,
+    `response.aggregations.${AGGREGATED_ANOMALIES}.buckets`,
+    []
+  );
+  let anomalies = [] as AnomalyData[];
+  let endTimeOffset = 0;
+  let prefix = '';
+  let dateFormat = '';
+
+  switch (anomalyAgg) {
+    case ANOMALY_AGG.DAILY: {
+      endTimeOffset = DAY_IN_MILLI_SECS;
+      prefix = '';
+      dateFormat = 'MM/DD/YY';
+      break;
+    }
+    case ANOMALY_AGG.WEEKLY: {
+      endTimeOffset = WEEK_IN_MILLI_SECS;
+      prefix = 'Week of ';
+      dateFormat = 'MM/DD/YY';
+      break;
+    }
+    case ANOMALY_AGG.MONTHLY: {
+      // Approximate end time since months don't have even days
+      endTimeOffset = DAY_IN_MILLI_SECS * 30;
+      prefix = '';
+      dateFormat = 'MMM YYYY';
+    }
+  }
+  const plotTimeOffset = endTimeOffset / 2;
+
+  resultBuckets.forEach((bucket: any) => {
+    const timestamp = get(bucket, 'key');
+    let maxGrade = get(bucket, `${MAX_ANOMALY_AGGS}.value`);
+    anomalies.push({
+      anomalyGrade: maxGrade !== null ? maxGrade : 0,
+      //@ts-ignore
+      confidence: undefined,
```

Comment:
Frontend doesn't want to show.

Reply:
could you explain why?

Reply:
UX & product decision. It was considered that the confidence was more relevant in the real-time setting as the detector is ingesting more and more data. In the historical context, it may be fed years of data, so the confidence will almost always be really high and thus irrelevant when looking at the data.